### PR TITLE
100 change to ssh

### LINKED ISSUE #100: Address changing authentication requirements in Lesson 3
GitHub is changing requirements for command line interaction on August 13, 2021.   While practicing for an upcoming workshop, I found using the instructions under Pushing Changes in Lesson 3 prompted am email Deprecation Notice email from GitHub which may disrupt the flow of a lesson.  I recommend addressing the new requirements outlined here https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/

### PR DESCRIPTION
**Pull request by emcaulay**
Resolves #100. 

This is a major revision. 

In order to address the change in authentication required by GitHub and to implement The Carpentries selected approach (SSH), I needed to move some of the Git configuration into Episodes 2 and 3 of the Lesson. There are revisions to the 

- prerequisites for the lesson
- the setup for the lesson
- Episode 2
- Episode 3

I appreciate all reviewers for their time in reviewing this work and welcome suggestions and corrections.

### CODE REVIEW

**Review by doingarchives — APPROVED** [review]

**Review by doingarchives — CHANGES_REQUESTED** [review]
Would it be possible to use a consistent and sequential naming / numbering scheme for infographics? This would facilitate future maintenance tasks. Open to many approaches on the name - 'image001lcgit'?

See ISO 5127:2017(en)

3.2.1.04
image
retinal pattern formed by light reflected or transmitted by external stimuli, whose impression is completed by the physiological mechanisms and mental processes that affect visual perception (3.1.1.33)

Note 1 to entry: In graphic technology, the term is commonly used to identify any picture, drawing, illustration, graphic, text or other reproduction, visible to the human eye, that portrays the original in the proper form, colour and perspective.
[SOURCE: ISO 12637-1:2006, definition 31]

Note 2 to entry: “Image” generally refers to any kind of viewing perception, not only to that of pure pictorial content. In this general capacity, “image” also includes e.g. pure text (3.2.1.05) which itself consists solely of graphic/writing (3.1.11.01)characters (3.1.4.02). Thus, in certain data processing (3.1.12.01) operations, processing and display (3.1.11.14) of pure text nevertheless takes place as “image processing” (e.g. in an “image catalogue”).

Note 3 to entry: The ‘external stimuli’ can, when applying certain technical operations like photographing (3.3.5.15), be fixed onto a data medium (3.1.1.39) and thus a document (3.1.1.38) be created from the image.

Note 4 to entry: See also “picture” (3.4.7.51) and ISO 21127:2014, classes “E 36” and“E 38”.

**Review by doingarchives — CHANGES_REQUESTED** [review]
Kindly see prior comment.

**Review by doingarchives — CHANGES_REQUESTED** [review]
Not fond of unnecessary capitalisation - is 'Git' a trademarked term? I've also considered the 'Dracula' example a bit tone deaf and socially fraught. ISO / IT terms that are neutral would be better 'Your Name' / 'yourname@address.org'

**Review by doingarchives — CHANGES_REQUESTED** [review]
Suggest
with git is the language. Although some of the language used is ...

**Review by doingarchives — COMMENTED** [review]
See prior comments ...

**Review by doingarchives — CHANGES_REQUESTED** [review]
The text editor selection and specification has always been problematic in this lesson. There has to be an open source option available - I used BBEdit (not free) - a better way would be to discuss the topics and commands as if any text editor would work. Again we don't have useful data on instructor practices, I'm guessing there is a high degree of variation among instructors - they use what is on their machines used for the class.

**Review by doingarchives — CHANGES_REQUESTED** [review]
Savvy librarian (like Dracula) is socially awkward and value laden. General comments on the use of capitalization discussed elsewhere. Understand the SSH discussion is the main point here. Acronym should be translated in plain English before the letters SSH. The crypto details should be left for another course / topic ... it appears to be answering a need that is not real (lines 102-106). Drop 'Because generally speaking .... (line 115); also Your output is going (131) - this information is not really adding anything instructional (it is observational and individual experiences will vary confusingly). (line 163) This 'make a note of it' is contrary to many best practices - we were specific about text editors, why not secure password solutions - just avoid this if possible.

**Review by doingarchives — CHANGES_REQUESTED** [review]
Choice of a license should reflect the 'open' nature of Library Carpentries - something to the effect of unless there is a legal or economic requirement ... open Creative Commons licenses are always the better choice.

**Review by doingarchives — COMMENTED** [review]
See earlier comment about consistent and sequential text and image assets.

**Review by doingarchives — COMMENTED** [review]
See earlier comment about consistent / sequential naming of assets.

**Review by doingarchives — CHANGES_REQUESTED** [review]
Consider this text (from Apple) - something like this would work well for LC audiences

"If you’re new to using the command line and don’t anticipate using it much for editing, nano is probably your best choice. If you expect to spend a lot of time using the command-line environment, it’s probably worth learning vim. They have very different design philosophies, so spend some time with each of them to determine which works best for you."

See Apple Computer 2021. Use command-line text editors in Terminal on Mac. Last accessed 2021 08 04 at < https://apple.co/3lwkwcQ >.

**Comment by doingarchives** [comment]
A lot of quality work here ... my basic sense is that the technical point of the SSH authentication has become embedded in several text revisions with a broader scope. That said, especially with all the effort made to this point - some of the biases and pedagogical issues in these text passages can be fixed as well. Happy to socialize the conversation and lend a hand as needed.

**Comment by chendaniely** [comment]
Hi @doingarchives I'm helping @emcaulay, one of the newer maintainers for this lesson, and we're both a bit confused about some of the comments since they seem to reference specific parts of the changes but for some reason, we both don't see which specific parts you're pointing to.

One example is from https://github.com/LibraryCarpentry/lc-git/pull/106#pullrequestreview-722435598

What we both see is this:
image

and I can't seem to find a reference to another repo you're referring to which is making the additional fixes a bit hard to track down.


also, since @emcaulay is still working on this issue to incorporate your comments, can we re-open this PR?

**Comment by emcaulay** [comment]
@jt14den , can you help me with this pull request and review? thanks!

**Comment by emcaulay** [comment]
Thanks for the full review, @doingarchives. 

I will reply to each suggestion with a response and try to be clear about which comment I am reacting to.

**Comment by doingarchives** [comment]
Hello Daniel

Nice to meet you.
I’m happy to be collaborating on the revisions.

It might be better to fork the course and send you suggestions in a new branch.
I’ll have a bit of time to do that on Thursday.

Has Francois (or someone else at LC administration) added you to lc-git-maintainers?
I’ll try and be granular in my suggestions.

Christopher Felker JD PhD
twitter [dot][at]doingarchives
electronic mail doingarchives [at] gmail [dot] com

On Aug 4, 2021, at 17:34, Daniel Chen ***@***.***> wrote:

﻿
Hi @doingarchives I'm helping @emcaulay, one of the newer maintainers for this lesson, and we're both a bit confused about some of the comments since they seem to reference specific parts of the changes but for some reason, we both don't see which specific parts you're pointing to.

One example is from #106 (review)

What we both see is this:


and I can't seem to find a reference to another repo you're referring to which is making the additional fixes a bit hard to track down.

also, since @emcaulay is still working on this issue to incorporate your comments, can we re-open this PR?

File: _episodes/03-sharing.md [hunk 1]
```diff
@@ -95,6 +95,150 @@ origin  https://github.com/<your_github_username>/hello-world (push)
...
+
+We will run the list command (`ls`) to check what key pairs already exist on your computer. 
+In our command we use the `~` as the shorthand for "my home directory."
+
+~~~
+ls -al ~/.ssh
+~~~
+{: .language-bash}
+
+Your output is going to look a little different depending on whether or not SSH has ever been set up on the computer you are using.
```
Comment (chendaniely) [inline _episodes/03-sharing.md:131]: from @doingarchives's comment:

> Your output is going (131) - this information is not really adding anything instructional (it is observational and individual experiences will vary confusingly)

I think this part of the lesson is for the instructor to figure out who actually needs to create a key, and who already has one and can skip the next few commands on creating a key.
We can make this distinction more explicit here and maybe say something like "if you do have existing keys, skip the next section and go here: xxxx" and link to the part where everyone copies their `.pub` file to paste into GitHub.

**Comment by emcaulay** [comment]
Reopening so that I can work through these comments.

**Comment by jt14den** [comment]
@emcaulay Thanks! I can review later this week. Looks like a lot of good work!

**Comment by chendaniely** [comment]
Hi @doingarchives!
I'm not part of this repo. I just help out the maintainers as a whole without the "big green button" access :)
@emcaulay just asked for some help during the co-working session today I'm hosting to walk her through the github PR interface. I think we both usually see specific comments to specific lines in context so we were both a bit confused and she wanted to make sure I was seeing the same thing.

**Comment by emcaulay** [comment]
Topic 1 from @doingarchives review: Change to ISO standard for filenaming of images. 

Worth considering, but it is outside the scope of the purpose of this revision. This revision should focus on adjusting the lesson to support teaching SSH as the authentication to GitHub method as discussed in issue #100. 

I have created issue #107 to open discussion about this suggestion.

**Comment by emcaulay** [comment]
I can't tell which reference you are mentioning here since it's not tagged to a line of the lesson. I will come back to this comment if I can find out what you're highlighting. 

> Not thrilled by reference within a course to material covered in another course. In my view, there are learners in the audience that take specific courses (not sure what percentage - that would be useful data). For these learners, elliptical references can multiply their sense of frustration - something the text here is already anticipating. Also I find this passage to make assumptions about the learning process that are unfortunate (negative outcomes as normal). Again I'm open to assisting on this or socialising the conversation - the reference would work better if the instruction were internalized.

**Comment by emcaulay** [comment]
Capitalization is used not just for trademarks but for other proper names. In my research in preparing these revisions, I found that Git, the version control system, is capitalized. It is a thing and it has a name and it is a specific instance of a thing. 

My references for this analysis are:

Usage for the word "Git" to refer to the version control system in these two O'Reilly books:

- Loeliger, Jon and Matthew McCullough. _Version Control with Git_. (Sebastopol, CA: O'Reilly, 2012).
- Westby, Emma Jane Hogbin. _Git for Teams_. (Sebastopol, CA: O'Reilly, 2015). 

Discussed as part of several capitalization questions here: https://stackoverflow.com/questions/37418392/how-to-capitalize-xcode-git-git-and-head. 

Please also see that I'm aligning with the Software Carpentry usage as well: https://github.com/swcarpentry/git-novice/issues/505#issue-319273431. 



> Not fond of unnecessary capitalisation - is 'Git' a trademarked term?

**Comment by emcaulay** [comment]
More replies tomorrow -- I ran out of time for today. I will make revisions to remove use of "Savvy Librarian" as an example user. I will use "Your Name" instead.

File: _episodes/03-sharing.md (hunk 1 above)
Comment (doingarchives) [reply _episodes/03-sharing.md:131]: @chendaniely In my instructor experience - the availability of a user account is a set up issue ... the key should already be in the possession of the person taking the class. Often this is the case in most types of software training - I recognise @emcaulay is wrestling with this issue as far as where it goes in the lesson plan ... the point of the lesson is not to 'set up' Git for a user / learner but to demonstrate the typical use case for a professional. The somewhat repugnant 'savvy librarian' like the characters on moralistic fiction would already come to class with their Git user credentials. To elevate the lesson I think we should address the credential issue (and any security needs) as a set up item, then merely reference the password in the lesson body. One could add a - 'Oops!? No password, see the set up instructions you completed prior to the course.

COVID-19 and the learner caps in place at most UC institutional offerings means that for every registered user wanting a live demo - there are many others who do the lesson remotely. As this trend accelerates, the need to maintain lessons that are 'self contained' and 'self referential' is a significant policy priority.

Socialising the discussion and perhaps blocking out times and best channels is a great first step for the moderator / maintainer cadre. At UC San Diego Slack was always open so it was easier to be timely in responding. At Harvard (alas) Teams is the norm - and that is almost nowhere to be found in the Carpentries ecoverse.

File: _episodes/02-getting-started.md [hunk 2]
```diff
@@ -41,6 +129,17 @@ $ cd hello-world
 ~~~
 {: .language-bash }
 
+### Using Git
+
+One of the main barriers to getting started with Git is the language. Although some of the language used in Git is
```
Comment (emcaulay) [inline _episodes/02-getting-started.md:134]: I think this could be the section of the lesson that @doingarchives might have been referring to as having a negative tone. I agree, and I would like to change it, but I'm going to put that as a new issue. Focused here on getting the SSH authentication integrated into the lesson in time for the August 13 deadline.

**Comment by emcaulay** [comment]
I'm interested in your comment below, but it wasn't attached to a specific section of the episode, so I was kind of guessing about what you're talking about. I think I agree in principle. I've opened a new issue for this discussion. #110 

> Not thrilled by reference within a course to material covered in another course. In my view, there are learners in the audience that take specific courses (not sure what percentage - that would be useful data). For these learners, elliptical references can multiply their sense of frustration - something the text here is already anticipating. Also I find this passage to make assumptions about the learning process that are unfortunate (negative outcomes as normal). Again I'm open to assisting on this or socialising the conversation - the reference would work better if the instruction were internalized.

**Comment by emcaulay** [comment]
> @emacauley Thanks for this. Could we agree to consider changing the example address to this: "[yourname@domain.name](mailto:yourname@domain.name)" See ISO 5127 3.1.5.46 (test over on slack).

Yes, I will make those changes now.

File: _episodes/03-sharing.md (hunk 1 above)
Comment (emcaulay) [reply _episodes/03-sharing.md:131]: I also have instructor experience and am a librarian, and my experience differs from yours @doingarchives. I think set up is an integral part of this lesson. GitHub and private and public keys are not part of our target demographic's general activities. The goal of this lesson and the core LC workshop series is to enable the learner to adopt these technologies and I believe the Git and GitHub set up is sufficiently alien that it should be in the lesson not outside the lesson. The much more sophisticated audience and extensive lesson in the Software Carpentry program also has decided to include the set up within the lesson. 

As a separate matter, I have removed the use of "Savvy Librarian" as you requested.

Comment (emcaulay) [reply _episodes/03-sharing.md:131]: As far as "socialising the discussion," I think you mean have a meeting? But I'm not certain since that phrase is unfamiliar to me. 

I think since you were on-boarded, @chendaniely has taken over the role of Maintainers Coordinator across Library, Data, and Software Carpentry. He hosts 2 meetings a month (with 2 different time slots to accommodate the wide range of time zones). The notes and connection information are here: https://codimd.carpentries.org/maintainers?both


I am still learning the different methods of communication for the Carpentries Maintainers, but there is definitely:

- TopicBox: https://carpentries.topicbox.com/groups/maintainers
- Slack: swcarpentry.slack.com(i recognize you are saying you do not regularly use that)

File: _episodes/02-getting-started.md (hunk 2 above)
Comment (doingarchives) [reply _episodes/02-getting-started.md:134]: Understood - as a maintainer I like to see changes come from the user base (instructors / learners). The data we have on most granular subjects is sparse - I think substantive changes should be polled to either the advisory committee or a sample of users (or potential users). We'll not get statistical accuracy but we can triangulate some of the conversations. When the proposals are 'lateral' meaning maintainer to maintainer, I'll put 'eyes' (the emoji) on those pseudo issues and perhaps they can be adjudicated at a higher level (advisory) or with a poll of former learners / instructors and prospective learners/instructors.

Comment (doingarchives) [reply _episodes/02-getting-started.md:134]: The narrow technical SSH issue can be fixed well before the deadline. It was embedded in a more thoughtful (and necessary) conversation.

**Comment by emcaulay** [comment]
I don't know where 

"If you are stuck here, look at the set up instructions for this lesson. The best way to solve the problem is register for a free Git account".

is in this lesson. It doesn't sound familiar and I've gone through episodes 1 and 2 fairly carefully, but I'll open a new issue for this comment as well. I agree that I don't like the way those two sentences sound and it doesn't make much sense to me.

**Comment by doingarchives** [comment]
Hello Daniel

Awesome.
I olli at GitHub as a development environment primarily.

There is some confusion (not just on lc-git) but on another lesson I maintain (lc-data-intro-archives).
I own some of that because of multiple GitHub accounts.

For now, doingarchives on GitHub and Slack are the best channels for the maintenance work.
I’ll port the discussions to Slack.

Typical issues / pulls involve just a few lines or even words in the lesson.
The re considerations of several of the longer sections, and even proposed revisions might be better on Slack and/or a document like Etherpad.

Good to be working with you all. 

Christopher Felker
Global open learning, teaching + data about archives and their communities
@doingarchives

On 2021 04 08, at 18:17 , Daniel Chen ***@***.***> wrote:


Hi @doingarchives <https://github.com/doingarchives>!
I'm not part of this repo. I just help out the maintainers as a whole without the "big green button" access :)
@emcaulay <https://github.com/emcaulay> just asked for some help during the co-working session today I'm hosting to walk her through the github PR interface. I think we both usually see specific comments to specific lines in context so we were both a bit confused and she wanted to make sure I was seeing the same thing.

File: _episodes/02-getting-started.md [hunk 3]
```diff
@@ -1,29 +1,117 @@
...
-our own website.
+When we use Git on a new computer for the first time,
+we need to configure a few things. The basic elements of a configuration for Git are:
+
+*   your name and email address,
+*   what your preferred text editor is,
+*   set the name of your default branch (branches are an important component of Git that we will cover later)
+*   and that you want to use these settings globally (i.e. for every project).
+
+First, we will tell Git our user name and email. For this lesson, we will be interacting with [GitHub](https://github.com/) and so we want to use the same email address we used when we set up our GitHub account. If you are concerned about privacy, please review [GitHub's instructions for keeping your email address private][git-privacy].
```
Comment (kekoziar) [inline _episodes/02-getting-started.md:29]: On Line 29: 
the format is 
\[GitHub’s instructions for keeping your email address private]\(https://help.github.com/articles/keeping-your-email-address-private/)

File: _episodes/03-sharing.md [hunk 4]
```diff
@@ -19,69 +19,69 @@ keypoints:
...
+
+* Click "New Repository"
+
+Clicking `New Repository` will take you to a creation page with different options. For this workshop, we are not using any of the options available. 
+
+* Name your repository "hello-world."
+
+![The 'create a new repository' form on GitHub](../fig/github-repo-new-setup.png) 
 
-GitHub will ask if you want to add a README.md, license or a `.gitignore` file. Do not do any of that for now.
```
Comment (kekoziar) [inline _episodes/03-sharing.md:30]: I don't agree with removing this line 
> GitHub will ask if you want to add a README.md, license or a `.gitignore` file. Do not do any of that for now.

If someone includes any files on the remote when it's created, it will cause an error when the local repo is pushed.  Since we're setting up learners to succeed at one thing at a time, it's important that the remote repository is completely empty.

File: index.md [hunk 5]
```diff
@@ -4,7 +4,7 @@ root: .
 ---
 #### What We Will Try to Do
 
-**Begin** to understand and use Git/GitHub. You will not be an expert by the end of the class. You will probably not even feel very comfortable using Git. This is okay. We want to make a start but, as with any skill, using Git takes practice.
+**Begin** to understand and use Git/GitHub. You will not be an expert by the end of the class. You will probably not even feel very comfortable using Git. This is okay. We want to make a start but, as with any skill, using Git takes practice. We will be using what we learned in The Unix Shell to use Git on the command line for this lesson.
```
Comment (kekoziar) [inline index.md:7]: The SWC uses this language in its prerequisite section

> In this lesson we use Git from the Unix Shell. Some previous experience with the shell is expected, but isn’t mandatory.

maybe adapt that?

**Review by kekoziar — COMMENTED** [review]
All other changes, especially SSH, look good.

File: _episodes/02-getting-started.md (hunk 3 above)
Comment (emcaulay) [reply _episodes/02-getting-started.md]: Thank you so much -- excellent catch. I revised and did a new commit.

File: _episodes/03-sharing.md (hunk 4 above)
Comment (emcaulay) [reply _episodes/03-sharing.md:30]: I think you might not have noticed that I had a line about that above the screenshot, but I think you might be right that the clarity of that sentence is a good repetition. I will add it back in. 

This is what I did have in the lesson, but I think the two sentences you point out in this comment are worth keeping indeed. 
On line 35 of this lesson, I have: 

> Clicking `New Repository` will take you to a creation page with different options. For this workshop, we are not using any of the options available.

File: index.md (hunk 5 above)
Comment (emcaulay) [reply index.md:7]: I really like that sentence. 

That is exactly how I have taught these lessons in a workshop. We emphasize the relationship between the Unix Shell lesson and the Git lesson, and we use it as an opportunity to reflect back to the learners that they have made progress in accumulating new skills already.

File: _episodes/03-sharing.md (hunk 4 above)
Comment (kekoziar) [reply _episodes/03-sharing.md:30]: @emcaulay, Yes, I did lose that sentence.

File: _episodes/02-getting-started.md (hunk 3 above)
Comment (tobyhodges) [reply _episodes/02-getting-started.md]: For the sake of completeness, and because you may very well see both forms used elsewhere in Carpentries lessons (sometimes together in the same page/lesson), the 

```markdown
[GitHub's instructions for keeping your email address private][git-privacy]
```

syntax is also valid, if you include a link reference later in the file, e.g. 

```markdown
[GitHub's instructions for keeping your email address private][git-privacy]

[git-privacy]: https://help.github.com/articles/keeping-your-email-address-private/
```

It is standard practice to include all such link references at the very bottom of the Markdown file. These "reference-style" links in Markdown are particularly useful for URLs that are linked to in multiple locations throughout the document: that way, if the address of the target page changes, you only need to modify the URL in a single location to update all the relevant links.

To be clear, I am not suggesting that you need to take this approach here, but adding the information because it may be helpful when reading/contributing to other repositories.

Comment (ErinBecker) [reply _episodes/02-getting-started.md]: Thank you @tobyhodges! I remember being so confused by this when I started working with Carpentries lessons.

File: _episodes/03-sharing.md (hunk 1 above)
Comment (ErinBecker) [reply _episodes/03-sharing.md:133]: I agree with @emcaulay here about the overall goals for the lesson including helping learners with setup complications. This is a core part of the "meet the learner where they are" principle from Instructor Training.

**Comment by ErinBecker** [comment]
> Not thrilled by reference within a course to material covered in another course. In my view, there are learners in the audience that take specific courses (not sure what percentage - that would be useful data). For these learners, elliptical references can multiply their sense of frustration - something the text here is already anticipating. Also I find this passage to make assumptions about the learning process that are unfortunate (negative outcomes as normal). Again I'm open to assisting on this or socialising the conversation - the reference would work better if the instruction were internalized.

I don't know for sure for LC workshops, but for SWC workshops, there is a requirement to teach both the shell and git lessons. If that is also the case for LC, I am in favor of cross-referencing lesson materials to encourage learners to make connections with what they have already learned. It may very well be true that there are learners who use the lessons outside of workshops, but our target audience is workshop learners and we should match the materials to that audience. 🙂

File: _episodes/03-sharing.md [hunk 6]
```diff
@@ -95,6 +97,150 @@ origin  https://github.com/<your_github_username>/hello-world (push)
 ~~~
 {: .output}
 
+## SSH Background and Setup
+We still need to do a little more setup before we can actually connect to this remote repository.
+We need to set up a way for our local computer to authenticate with GitHub so that GitHub recognizes our computer as belonging to the same person who owns the GitHub repository. 
+
+We will use SSH as our authentication method. SSH stands for Secure Shell Protocol.  SSH is a cryptographic network protocol that allows secure communication between computers using an otherwise insecure network.
```
Comment (tobyhodges) [inline _episodes/03-sharing.md:104]:
```suggestion
We will use SSH as our authentication method. SSH stands for Secure SHell protocol.  SSH is a cryptographic network protocol that allows secure communication between computers using an otherwise insecure network.
```

I found it distracting that the abbreviation of Secure Shell Protocol is not "SSP," and I think this tweak to the cases of a coupl characters helps with this. I do not feel strongly either way - feel free to reject this suggestion if you do not like it.

File: _episodes/03-sharing.md (hunk 4 above)
Comment (tobyhodges) [reply _episodes/03-sharing.md:30]: I am in favour of adding the removed line back in too. Any files added to the GitHub repository when it is created here will cause trouble later in the lesson, when the learner first tries to sync their local and remote versions, and I think the redundancy of including both these sentences is worth it to further guard against that.

**Review by tobyhodges — APPROVED** [review]
This is excellent, and I am delighted to see the coordination between `lc-git` and `swcarpentry/git-novice` to make it happen. In my view it is ready to merge, but I have made a couple suggestions of changes for minor improvement. If it is merged here, I will open a PR to propagate the _Secure Shell Protocol_ -> _Secure SHell protocol_ change over on the Software Carpentry lesson too.

**Comment by tobyhodges** [comment]
> I don't know for sure for LC workshops, but for SWC workshops, there is a requirement to teach both the shell and git lessons. If that is also the case for LC, I am in favor of cross-referencing lesson materials to encourage learners to make connections with what they have already learned.

It is my understanding (from [the Library Carpentry Workshop Overview](https://librarycarpentry.org/lc-overview/#curricula)) that an LC workshop must teach at least three of the four core lessons. Technically, that does leave the possibility that instructors might teach the Git lesson without first teaching the Shell lesson, but I do not think that would be likely to happen, especially if the language added in this PR is merged:

> In this lesson we use Git from the Unix Shell. Some previous experience with the shell is expected, but is not mandatory.

File: _episodes/02-getting-started.md (hunk 3 above)
Comment (emcaulay) [reply _episodes/02-getting-started.md]: I think I will come back and improve the linking syntax and adding the link references to the bottom of the Markdown file in a separate pull request. I need to learn this standard practice, but it looks like what I used is functional for the time being.

Comment (tobyhodges) [reply _episodes/02-getting-started.md]: Absolutely! Indeed, I meant to say that, if using the reference-style syntax for links, then it is standard practice to put the references at the end of the file. But plenty of projects, including many Carpentries lessons, use the "inline" syntax you are using here. I agree it is a minor thing and best left to a separate PR, if you want to do it at all.

**Comment by ErinBecker** [comment]
Thank you @emcaulay for the work that you've put into this PR. I've reviewed, along with @tobyhodges, @chendaniely, and @kekoziar, and this looks great! Although there are some unresolved comments in the conversation thread, these are outside of the scope of this PR and can be dealt with in separate issues/PRs. Since the security issue that this PR is addressing is due to land on the 13th, I'm going to go ahead and merge this (with thanks!!) to keep the lessons from breaking. Happy to continue some of these broader conversations in other threads.

**Comment by doingarchives** [comment]
@emcaulay Kindly see Slack for additional discussion. Not seeing connection between Update link syntax throughout lesson ISS#112 ... looking for markdown reference(s).
